handler/oauth2/flow_resource_owner_test.go

```diff
@@ -38,31 +38,54 @@ func TestResourceOwnerFlow_HandleTokenEndpointRequest(t *testing.T) {
 			description: "should fail because not responsible",
 			expectErr:   fosite.ErrUnknownRequest,
 			setup: func() {
-				areq.GrantTypes = fosite.Arguments{"123"}
+				areq.EXPECT().GetGrantTypes().Return(fosite.Arguments{""})
```

Comment:
No reason, I've copied the setup from the flow_client_credentials tests, thinking it would be better if they used the same checks. I can change it back to 123 but thought the test would need to fail either way. Let me know what you think is best.